catalog/resource_grants.go

```diff
@@ -93,16 +94,27 @@ type securableMapping map[string]map[string]bool
 
 // reuse ResourceDiff and ResourceData
 type attributeGetter interface {
-	Get(key string) any
+	GetRawConfig() cty.Value
 }
 
 func (sm securableMapping) kv(d attributeGetter) (string, string) {
+	rawConfig := d.GetRawConfig()
+	if rawConfig.IsNull() {
+		return "unknown", "unknown"
+	}
+	rawConfigValues := rawConfig.AsValueMap()
 	for field := range sm {
-		v := d.Get(field).(string)
-		if v == "" {
+		value := rawConfigValues[field]
+		if value.IsNull() {
 			continue
 		}
-		return field, v
+		// We need this check because:
+		// "AsString returns the native string from a non-null, non-unknown cty.String value, or panics if called on any other value."
+		if !value.IsKnown() {
+			// This is the case when the new value isn't known but the field is required for validation
+			return field, "unknown"
+		}
+		return field, value.AsString()
 	}
 	return "unknown", "unknown"
```

Comment:
What should we do in this case?

Reply:
In the case of "unknown", "unknown" we fail at plan which is expected right?